crates/ra_hir_ty/src/tests/coercion.rs

```diff
@@ -275,12 +275,14 @@ fn test(i: i32) {
     [70; 147) 'match ...     }': &[i32]
     [76; 77) 'i': i32
     [88; 89) '2': i32
+    [88; 89) '2': i32
```

Comment:
Why the duplicates?

Reply:
One is the pattern (which was previously inferred 'backwards' in this case), one is the expression inside the pattern.